app/assets/javascripts/components/buttons/generic-button-component.js

```diff
@@ -0,0 +1,41 @@
+ManageIQ.angular.app.component('genericButtonComponent', {
+  bindings: {
+    angularForm: '=',
+    modelCopy: '=',
+    model: '=',
+    redirectUrl: '=?',
+    newRecord: '<',
+    entity: '=?',
+    entityName: '=?',
+    onSaveClicked: '&?',
+    onAddClicked: '&?',
+  },
+  controllerAs: 'vm',
+  controller: genericButtonController,
+  templateUrl: '/static/buttons/generic_button_component.html.haml',
+});
+
+genericButtonController.$inject = ['miqService', '$window'];
+
+function genericButtonController(miqService) {
+  var vm = this;
+
+  vm.$onInit = function() {
+    vm.saveable = miqService.saveable;
+
+    vm.resetClicked = function() {
```

Comment:
Yes, but it's too evolved :). I have a BZ I'm working on that we need to start putting the form buttons back in paging toolbar, the plan was to have a set of 2 components, one of which is part of the form, and so knows what should be grey/blue/etc. and talks to the other one which is not part of the form but part of the bottom toolbar, and just shows the buttons.

With the design I had in mind ,this is trivial. Once you start adding behaviour like restting the form and moving elsewhere to the mix, it becomes ..less trivial.